pkg/controllers/updaterun/initialization.go

```diff
@@ -543,6 +532,52 @@ func (r *Reconciler) recordOverrideSnapshots(ctx context.Context, placementKey t
 	return nil
 }
 
+// getResourceSnapshotObjs retrieves the list of resource snapshot objects from the specified ResourceSnapshotIndex.
+// If ResourceSnapshotIndex is unspecified, it returns the list of latest resource snapshots.
+func (r *Reconciler) getResourceSnapshotObjs(ctx context.Context, updateRunSpec *placementv1beta1.UpdateRunSpec, placementName string, placementKey types.NamespacedName, updateRunRef klog.ObjectRef) ([]placementv1beta1.ResourceSnapshotObj, error) {
```

Comment:
placementName can be derived from placementKey, no need to pass in both. updateRunRef can be derived from updateRun obj. This signature only need ctx, updateRun obj (not just spec) and placementKey.